### DISCUSSION TITLE
Application Scoped DataLoader

### DISCUSSION THREAD

**Discussion by gilteplitsky** [Netflix/dgs-framework#763] [Q&A] [answered]
Is there a way to configure the framework to reuse DataLoaders so they will be application scoped and not request scoped?

As much as I understand the DgsDataLoaderProvider is creating new DataLoaders per request and with application scoped data loaders multiple http requests will be batched to perform a single DB query.

Thanks

**Comment by paulbakker** [comment, marked as answer]
DataLoaders are designed to batch (database/backend) calls within a single request. This is required because a field datafetcher, which might need to load from an external datasource,  will be invoked for each instance of its parent.
E.g. if I have a DataLoader for `Product.price`, and I load 500 products from a `Query.products` query, the `Product.price` datafetcher will be called 500 times. If `Product.price` would call out to a database, we would need to batch that.

Doing something similar for multiple requests adds many complications. Just to give some examples:
1. How do we even know another request will come in that has similar needs
2. It will negatively impact response times if requests have to wait to "align" so that they can be batched
3. Authz; does each request have the same permissions to the same data

Although there might be answers to these questions for highly specialized scenarios, the mechanism won't work in most cases. For scenarios it would work, it would probably be better to leverage caching instead of trying to batch.

**Reply by gilteplitsky** [reply]
thanks for the explanation
